src/invert/laplace/impls/pcr/pcr.cxx

```diff
@@ -0,0 +1,1137 @@
+        for (int kz = 0; kz < nmode; kz++)
+          bcmplx(kz, ix - xs) = k1d[kz];
+      }
+
+      // Get elements of the tridiagonal matrix
+      // including boundary conditions
+      BOUT_OMP(for nowait)
+      for (int kz = 0; kz < nmode; kz++) {
+        BoutReal kwave = kz * 2.0 * PI / (coords->zlength()); // wave number is 1/[rad]
+        tridagMatrix(&a(kz, 0), &b(kz, 0), &c(kz, 0), &bcmplx(kz, 0), jy,
+                     kz,    // True for the component constant (DC) in Z
+                     kwave, // Z wave number
+                     global_flags, inner_boundary_flags, outer_boundary_flags, &Acoef,
+                     &C1coef, &C2coef, &Dcoef,
+                     false); // Don't include guard cells in arrays
+      }
+    }
+
+    // Solve tridiagonal systems
+    //cr->setCoefs(a, b, c);
+    //cr->solve(bcmplx, xcmplx);
+    cr_pcr_solver(a,b,c,bcmplx,xcmplx);
+
+    // FFT back to real space
+    BOUT_OMP(parallel)
+    {
+      /// Create a local thread-scope working array
+      auto k1d = Array<dcomplex>((localmesh->LocalNz) / 2 +
+                                 1); // ZFFT routine expects input of this length
+
+      const bool zero_DC = global_flags & INVERT_ZERO_DC;
+
+      BOUT_OMP(for nowait)
+      for (int ix = xs; ix <= xe; ix++) {
+        if (zero_DC) {
+          k1d[0] = 0.;
+        }
+
+        for (int kz = zero_DC; kz < nmode; kz++)
+          k1d[kz] = xcmplx(kz, ix - xs);
+
+        for (int kz = nmode; kz < (localmesh->LocalNz) / 2 + 1; kz++)
+          k1d[kz] = 0.0; // Filtering out all higher harmonics
+
+        irfft(std::begin(k1d), localmesh->LocalNz, x[ix]);
+      }
+    }
+  }
+
+  checkData(x);
+
+  return x;
+}
+
+Field3D LaplacePCR::solve(const Field3D& rhs, const Field3D& x0) {
+  TRACE("LaplacePCR::solve(Field3D, Field3D)");
+
+  ASSERT1(rhs.getLocation() == location);
+  ASSERT1(x0.getLocation() == location);
+  ASSERT1(localmesh == rhs.getMesh() && localmesh == x0.getMesh());
+
+  Timer timer("invert");
+
+  Field3D x{emptyFrom(rhs)}; // Result
+
+  // Get the width of the boundary
+
+  // If the flags to assign that only one guard cell should be used is set
+  int inbndry = localmesh->xstart, outbndry = localmesh->xstart;
+  if ((global_flags & INVERT_BOTH_BNDRY_ONE) || (localmesh->xstart < 2)) {
+    inbndry = outbndry = 1;
+  }
+  if (inner_boundary_flags & INVERT_BNDRY_ONE)
+    inbndry = 1;
+  if (outer_boundary_flags & INVERT_BNDRY_ONE)
+    outbndry = 1;
+
+  int nx = xe - xs + 1; // Number of X points on this processor
+
+  // Get range of Y indices
+  int ys = localmesh->ystart, ye = localmesh->yend;
+
+  if (localmesh->hasBndryLowerY()) {
+    if (include_yguards)
+      ys = 0; // Mesh contains a lower boundary and we are solving in the guard cells
+
+    ys += extra_yguards_lower;
+  }
+  if (localmesh->hasBndryUpperY()) {
+    if (include_yguards)
+      ye = localmesh->LocalNy -
+           1; // Contains upper boundary and we are solving in the guard cells
```

Comment:
Might want to move the comment to the previous line so clang-format does do this :)